ledger/src/blockstore_processor.rs

```diff
@@ -1585,8 +1601,45 @@ pub mod tests {
         trees::tr,
     };
 
+    // Convenience wrapper to optionally process blockstore with Secondary access.
+    //
+    // Setting up the ledger for a test requires Primary access as items will need to be inserted.
+    // However, once a Secondary access has been opened, it won't automaticaly see updates made by
+    // the Primary access. So, open (and close) the Secondary access within this function to ensure
+    // that "stale" Secondary accesses don't propagate.
+    fn test_process_blockstore_with_access(
+        genesis_config: &GenesisConfig,
+        blockstore: &Blockstore,
+        opts: ProcessOptions,
+        process_with_secondary: bool,
```

Comment:
I think we can just directly pass the `AccessType` here and that will remove the `if/else` block

Reply:
Might need to plumb that through in the rest of the spots where the bool is currently used though, but I think `enums` are more readable anyways